src/functions/retroGen.ts

```diff
@@ -16,24 +17,23 @@ const retroGen: Handler = async (event: any, context?: Context, callback?: Callb
         console.error("ERROR: event is not defined.");
         return;
     }
-
-    const s3BucketService: S3BucketService = Injector.resolve<S3BucketService>(S3BucketService);
     const retroService: RetroGenerationService = Injector.resolve<RetroGenerationService>(RetroGenerationService);
     const retroUploadPromises: Array<Promise<ManagedUpload.SendData>> = [];
+    const sharepointAuthenticationService = new SharePointAuthenticationService();
+    const sharePointService = new SharePointService();
 
     event.Records.forEach((record: any) => {
         const visit: any = JSON.parse(record.body);
         const retroUploadPromise = retroService.generateRetroReport(visit)
-            .then((generationServiceResponse: { fileName: string, fileBuffer: Buffer}) => {
-
-                return s3BucketService.upload(`cvs-retro-reports-${process.env.BUCKET}`, generationServiceResponse.fileName, generationServiceResponse.fileBuffer)
-                    .then((result: any) => {
-                        return result;
-                    });
-            })
-            .catch((error: any) => {
-                console.log(error);
-            });
+        .then(async (generationServiceResponse: { fileName: string, fileBuffer: Buffer }) => {
+            const tokenResponse = await sharepointAuthenticationService.getToken();
```

Comment:
Need to be careful with error handling on remote services when using async/await - it will always return something, but it will ignore errors. Need to handle remote service failures somehow.